app/services/past_hearings_fetcher.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class PastHearingsFetcher < ApplicationService
+  attr_reader :prosecution_case_id
+
+  def initialize(prosecution_case_id)
+    @prosecution_case_id = prosecution_case_id
+  end
+
+  def call
+    hearing_summaries = Array(ProsecutionCase.find(prosecution_case_id).body["hearingSummary"]).map do |hearing_summary_data|
+      HmctsCommonPlatform::HearingSummary.new(hearing_summary_data)
+    end
+
```

Comment:
Hi, it could do with refactoring like the hearing_summaries being a separate method. One of the reasons is readability. I think the code should work well.